bsonutil/indexes.go

```diff
@@ -48,31 +85,45 @@ func ConvertLegacyIndexKeys(indexKey bson.D, ns string) {
 	originalJSONString := CreateExtJSONString(indexKey)
 	for j, elem := range indexKey {
 		switch v := elem.Value.(type) {
-		case int32, int64, float64:
-			// Only convert 0 value
+		case int:
 			if v == 0 {
 				indexKey[j].Value = 1
 				converted = true
 			}
+		case int32:
+			if v == int32(0) {
+				indexKey[j].Value = int32(1)
+				converted = true
+			}
+		case int64:
+			if v == int64(0) {
+				indexKey[j].Value = int64(1)
+				converted = true
+			}
+		case float64:
+			if math.Abs(v - float64(0)) < epsilon{
+				indexKey[j].Value = float64(1)
+				converted = true
+			}
 		case primitive.Decimal128:
 			// Note, this doesn't catch Decimal values which are equivalent to "0" (e.g. 0.00 or -0).
```

Comment:
Didn't you have a way of doing this with `math/big` in a previous commit?

Reply:
Ah, I saw the comment that we are not going to convert values such as ` Decimal values which are equivalent to "0"`. Not sure if that's a requirement? @tfogo Do you think the math/big is a better solution? I can put that back. Thx.

Reply:
Originally, I had a complex solution for it. But David suggested not to worry about that edge case since it's probably so rare. I think he thought it wasn't worth the extra complexity. But the math/big solution is simple enough that I think it would be fine to add it. What do you think?

Reply:
Yea, I agree. I will revert to the previous solution then!

Reply:
Yea, I agree. I will revert to the previous solution then!